frontend/src/components/SignIn.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+const SignIn = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [username, setUsername] = useState('');
+  const [isSignUp, setIsSignUp] = useState(true); // Start with sign up mode
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (isSignUp) {
+      // Handle sign up
+      const response = await fetch(import.meta.env.VITE_BACKEND_URL + '/api/register', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, username, password }),
+      });
+      const data = await response.json();
+      console.log(data); // Handle response accordingly
+    } else {
+      // Handle login
+      const response = await fetch(import.meta.env.VITE_BACKEND_URL + '/api/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, password }),
+      });
+      const data = await response.json();
+      console.log(data); // Handle response accordingly
+    }
+  };
```

Comment:
In the long-term it's better to define API related logic separately from components, because sometimes multiple UI surfaces can implement the same behavior and therefore the API logic needs to be shared.

It's better to do this preemptively, because it's harder to refactor a bigger codebase later on